microsoft/testsuites/vm_extensions/azureperformancediagnostics.py

```diff
@@ -0,0 +1,187 @@
+            credential=platform.credential,
+            subscription_id=platform.subscription_id,
+            cloud=platform.cloud,
+            account_name=storage_account_name,
+            resource_group_name=resource_group_name,
+        )
+
+        # Run VM Extension
+        extension = node.features[AzureExtension]
+        settings = {
+            "performanceScenario": "quick",
+            "traceDurationInSeconds": 5,
+            "srNumber": "",
+            "requestTimeUtc": datetime.now(timezone.utc).strftime("%Y%m%d%H%M%S%f"),
+            "storageAccountName": storage_account_name,
+            "resourceId": (
+                f"/subscriptions/{platform.subscription_id}/resourceGroups/"
+                f"{resource_group_name}/providers/Microsoft.Compute/virtualMachines"
+                f"/{node.name}"
+            ),
+        }
+
+        protected_settings = {
+            "storageAccountName": storage_account_name,
+            "storageAccountKey": account_credential.get("account_key"),
+        }
+
+        extension_result = extension.create_or_update(
+            name="AzurePerformanceDiagnosticsLinux",
+            publisher="Microsoft.Azure.Performance.Diagnostics",
+            type_="AzurePerformanceDiagnosticsLinux",
+            type_handler_version="1.0",
+            auto_upgrade_minor_version=True,
+            settings=settings,
+            protected_settings=protected_settings,
+        )
+
+        assert_that(extension_result["provisioning_state"]).described_as(
+            "Expected the extension to succeed"
+        ).is_equal_to("Succeeded")
+
+        # Verify report was created and uploaded to the storage account
+        self._container_client = get_or_create_storage_container(
+            credential=platform.credential,
+            subscription_id=platform.subscription_id,
+            cloud=platform.cloud,
+            account_name=storage_account_name,
+            container_name="azdiagextnresults",
+            resource_group_name=resource_group_name,
+        )
+
+        blob_iter = self._container_client.list_blobs(
+            name_starts_with="PerformanceDiagnostics"
+        )
+
+        report_exists = False
+        for _ in blob_iter:
+            report_exists = True
+
+        assert_that(report_exists).described_as(
+            "Expected to find a report in the storage account, but no report was found"
+        ).is_true()
+
+        # Delete VM Extension
+        extension.delete("AzurePerformanceDiagnosticsLinux")
+
+        assert_that(
+            extension.check_exist("AzurePerformanceDiagnosticsLinux")
+        ).described_as(
+            "Found the VM Extension still unexpectedly exists on the VM after deletion"
+        ).is_false()
+
+    def _is_supported_linux_distro(self, node: Node) -> bool:
+        is_supported = False
+        version = node.os.information.version
+        major_version = version.major
+        if type(node.os) == CentOs:
```

Comment:
If it's a big matrix, this part still can be data driven from an array like before.

Reply:
Oops. Good point. Thank you. Fixed.